components/MapView.js

```diff
@@ -371,15 +371,6 @@ class MapView extends React.Component {
     return { provider: this.props.provider };
   }
 
-  componentDidMount() {
-    const { region, initialRegion } = this.props;
-    if (region && this.state.isReady) {
-      this.map.setNativeProps({ region });
-    } else if (initialRegion && this.state.isReady) {
```

Comment:
For google+ios, this will cause a region calculation that is incorrect. For mapkit+ios it will cause the map to redundantly set the region because it will happen in `_onLayout` anyway.